packages/editor/src/components/post-actions/set-as-homepage.js

```diff
@@ -0,0 +1,216 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useMemo, useRef } from '@wordpress/element';
+import {
+	Button,
+	__experimentalText as Text,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { store as noticesStore } from '@wordpress/notices';
+
+/**
+ * Internal dependencies
+ */
+import { getItemTitle } from '../../dataviews/actions/utils';
+
+const SetAsHomepageModal = ( { items, closeModal } ) => {
+	const [ item ] = items;
+	const pageTitle = getItemTitle( item );
+	const { showOnFront, currentHomePage, isSavingSiteSettings } = useSelect(
+		( select ) => {
+			const { getEntityRecord, isSavingEntityRecord } =
+				select( coreStore );
+			const siteSettings = getEntityRecord( 'root', 'site' );
+			const currentHomePageItem = getEntityRecord(
+				'postType',
+				'page',
+				siteSettings?.page_on_front
+			);
+			return {
+				showOnFront: siteSettings?.show_on_front,
+				currentHomePage: currentHomePageItem,
+				isSavingSiteSettings: isSavingEntityRecord( 'root', 'site' ),
+			};
+		}
+	);
+	const isPageDraftRef = useRef( item.status === 'draft' );
+	const isPageDraft = isPageDraftRef.current;
```

Comment:
Can you explain why we are having to store this value in a ref?

Reply:
I did this as the `item.status` changes if a draft page is selected as the homepage, so this means we can reference the initial value rather than allowing this to change when the page is published. This prevents a re-render on the UI changes for draft vs published pages, but perhaps this can be removed if the loading state is handled differently (as in [your comment here](https://github.com/WordPress/gutenberg/pull/65426#discussion_r1853642346)).

Reply:
It definitely feels like a crutch. 

I'm also wondering if it's possible to use the `batch` endpoint to perform the status change and the site settings change in a single request. That might avoid any pre-mature re-render?